### PR TITLE
LateNight :: optional Big spinnies, various improvements

### PR DESCRIPTION
**Pull request by ronso0**
- [x] choose Spinny, CoverArt or nothing (darker background makes Spinny indicator easier to recognize)
- [x] choose big or small Spinny/CoverArt, from ywwg/latenight-biggerspinny
- [x] add FX3/FX4 buttons to master mixer & Mic/Aux units
- [x] clean up tool bar
- [x] simplify skin setting buttons templates
- [x] increase minimum window size to 1366x768px (1302,668px to consider Ubuntu's vertical taskbar)
- [x] fix beatsize spinboxes for qt5
- [x] persistent visibility connections in lower half (Mic/Aux, Samplers) [Bug 1779575](https://bugs.launchpad.net/mixxx/+bug/1779575)
- [x] add Broadcast button to tool bar [Bug 1776258](https://bugs.launchpad.net/mixxx/+bug/1776258)
- [x] add LateNight launch image
- [x] add Load/Save Sampler Bank buttons
- [x] fix control vinyl status light
- [x] fix sampler VU meter graphics (wrong size of vector templates was exported to correct size in skin)
- [x] show CUE point in samplers' overview [Bug 1781830](https://bugs.launchpad.net/bugs/1781830)

latenight-update
(just an illustration..of course big/small spinny is toggled simultaneously on all decks)

New LateNight launch image:
latenight-launch2

Please test and let's discuss if we increase the minimum size to 1366x768 or if we stick with 1280px screens.

### CODE REVIEW

**Comment by nopeppermint** [comment]
@ronso0 looks good, thanks for keeping small cover Art ;-)

**Comment by ronso0** [comment]
> looks good, thanks for keeping small cover Art ;-)

yeah, both big and small use cases are valid IMO.

**Comment by ywwg** [comment]
looks great thanks for picking this up!

**Comment by ronso0** [comment]
Done, please test.

**Comment by nopeppermint** [comment]
looks good

**Comment by nopeppermint** [comment]
maybe move "load/save sampler bank" to submenu "Sampler" ?

**Comment by ronso0** [comment]
We don't have a Samplers menu. Maybe we'll have one when we add more samplers or other sampler options, but for now I don't want to open another submenu just for this two buttons.

**Comment by ronso0** [comment]
Is anyone using controls vinyl? I don't so I can't test related controls and the status light

**Comment by nopeppermint** [comment]
>  that all buttons are dim when pressed.

which buttons are dimmed?

**Comment by ronso0** [comment]
all transport controls, rate buttons etc.
They are red/white text when active (i.e. HotCue is set) and get darker when you actually press them. IMO that doesn't match an action, i.e Play from cue, and I don't know of any hardware equivalent.
latenight-pressed

That's why I'd make the white text on 'active/set' buttons a tiny bit lesss bright, and the text/bg of such a pressed button brighter so pressed buttons stand up more.
latenight-pressed

**Comment by ywwg** [comment]
> @ywwg It feels strange that all buttons are dim when pressed. I'd like to change the icons so that the text is a bit brighter than unpressed. Do you mind?

that's fine with me! I'd also be ok with replacing the pixmaps with font-based buttons, the design is a little out of place (as is the red color)

**Comment by ywwg** [comment]
> Please test and let's discuss if we increase the minimum size to 1366x768 or if we stick with 1280px screens.

I think in the previous PR there was general consensus that 1366x768 would be ok.  My old 1280 laptop is long dead, which is what I designed the original to fit.  Did anyone else have an older machine that needed the smaller size?  (In general, I think we should save Shade for the smallest screens)

**Comment by ronso0** [comment]
rebased

**Comment by ronso0** [comment]
This is ready to merge.
The rest (fx mixmode buttons, buttons rework) will happen in another PR

**Comment by ronso0** [comment]
> preview deck text is cut off on my machine

don't know why this happens, font is included and shouldn't vary in size.
I replaced fixed size with `min,min` policy. Can you test again?

**Comment by ywwg** [comment]
looks good now.  I do have the accessibility feature "large fonts" enabled, at 125%, and this sometimes causes issues like these.

**Comment by ronso0** [comment]
great! so this can be merged.

**Comment by ywwg** [comment]
I think so, @daschuer any comments?

**Comment by nopeppermint** [comment]
@ronso0  why do you remove hover from star rating in decks?

**Comment by ronso0** [comment]
> @ronso0 why do you remove hover from star rating in decks?

After quite a while I used LateNight for an actual prep session. Check new tracks, sort them into crates or just delete them, rate them! This is were I found the hover effect disturbing, (=unexpected) in LateNight. Can't tell why I implemented it in the first place.. it doesn't match the skin anyway and it's not significant enogh to help, it's just disturbing IMO.
The actual star widget has a hover effect itself, so I removed it.

**Comment by daschuer** [comment]
The minimum width has been increased. Now it does not fit next to the Ubuntu Unity task bar on my 1366x768 screen. This is not a strong requirement, but if we can make it fit again, it would be nice.

**Comment by daschuer** [comment]
I found myself repeated clicking on the top right gear button for skin settings. 
Can we move the settings button there? We may also, or alternatively replace the gear button for the beatgrid with something else, a pencil icon, for "Edit beatgride" or something.

**Comment by ronso0** [comment]
> The minimum width has been increased. Now it does not fit next to the Ubuntu Unity task bar on my 1366x768 screen

Good point! So we should update the [skin_guidelines](https://www.mixxx.org/wiki/doku.php/skin_guidelines#the_optimal_size_for_skins) and add Ubuntu to the table.

> Can we move the settings button there?

Nope, the space above decks is reserved for collapsible waveforms only.
I can replace the gear with a pencil or a wrench, like I suggested for Maintenance in lib-redesign branch. Also I'll try to add the gear icon to the Settings button. It makes sense to have this consistent across skins.

**Comment by daschuer** [comment]
Yes, thanks. My vote would be a pencil, because a wrench might be scared double left handed people. :-p

File: res/skins/LateNight/skin.xml [hunk 1]
```diff
@@ -86,9 +88,35 @@
 
   <!-- MinimumSize should not be an exact monitor resolution. There needs
   to be space for the title bar or other chrome at full screen -->
-  <MinimumSize>1270,550</MinimumSize>
+  <MinimumSize>1344,668</MinimumSize>
```
Comment (daschuer) [inline res/skins/LateNight/skin.xml]: 1302,668 
for Full HD Displays running Ubunty Unity.

Comment (daschuer) [reply res/skins/LateNight/skin.xml]: Than the EQ buttons are cutted with the Skin preferences visible though.
But the Mixxx is usable on my device now. 
If we consider the "preferences open" state as exceptional it should work.

**Comment by daschuer** [comment]
There is only the pending size issue, the rest LGTM

**Comment by ronso0** [comment]
> My vote would be a pencil, because a wrench might be scared double left handed people. :-p

ha! I know what you mean..and was curious how to translate the german phrase. dict.leo.org says
"folks how are all thumbs"
so let's remove the gear as well, and the faders, all that stuff that MUST be something technical

File: res/skins/LateNight/skin.xml (hunk 1 above)
Comment (ronso0) [reply res/skins/LateNight/skin.xml]: 1366x768 is just HD (720+ lines)
actually I never really cared about squeezed skins with Pref pane open.
as long as the skin fits in its regular state where fine I think.

**Comment by Be-ing** [comment]
The minimum width and height are too wide to fit my 3840 x 2160 screen now (using Qt5's automatic scaling) :/

**Comment by Be-ing** [comment]
I tried cherry-picking d8b5714 to 2.1 but without the skin settings menu, there isn't enough horizontal space in the toolbar...

**Comment by ronso0** [comment]
> there isn't enough horizontal space in the toolbar...

Whaaat?
latenight_toolbarspacer

It's not realistic to test all skins with X screen resolutions and if automatic scaling fails so badly with 4K, we should bring back the possibility to opt out IMO

**Comment by Be-ing** [comment]
There is enough horizontal space in the toolbar for the broadcast icon in 2.2, but 2.1 has more buttons in the toolbar because it does not have the skin settings menu.

Automatic scaling works well. Why was the minimum size increased in this PR? LateNight worked fine on my screen before. The minimum size should be as small as possible, not just to fit in the widest variety of screen sizes, but also so Mixxx can be run more comfortably in a window that is not full screen.

**Comment by ronso0** [comment]
LateNight 2.1.0 Besides the question why you would import the broadcast button into 2.1, I see there's enough space for it at minimal window size:
latenight-2 1 0


> The minimum width and height are too wide to fit my 3840 x 2160 screen now (using Qt5's automatic scaling) :/

Please post a screenshot to show what is squeezed. Or is it just the warning in Prefences > Interface?
And yes, I consider Qt Scaling failing if it can't put a skin designed for 1366px on a 4K screen.

The minimum size was increased after we (incl. you) decided to do so in #1734.
This way we can have bigger spinnies AND a usable artist/title box.

> also so Mixxx can be run more comfortably in a window that is not full screen.

That's why I decreased it after @daschuer noticed it wouldn't fit next to the Ubuntu taskbar.

**Comment by ronso0** [comment]
Damn, I now notice this should actuallly go to 2.2 to have consitent Spinny/CoverArt controls.
@Be-ing @daschuer Do you agree? If yes, how to merge this into 2.2?

Also I'll soon open a small-fix PR to clean up the spinny COs in Deere so they really match those implemented here that should go to 2.2, as well.

**Comment by daschuer** [comment]
> Edit just pulled 2.2 and noitce this is already included...?

Yes, currently 2.2 and master are the same.

**Pull request merged**